server/prompt.go

```diff
@@ -114,7 +115,7 @@ func ChatPrompt(tmpl string, system string, messages []api.Message, window int,
 	for _, msg := range messages {
 		switch strings.ToLower(msg.Role) {
 		case "system":
-			if p.System != "" || p.Prompt != "" || p.Response != "" {
+			if !p.first && (p.System != "" || p.Prompt != "" || p.Response != "") {
```

Comment:
Let's check `len(p)` instead?

Reply:
I think we also need to check that it's equivalent to the provided system prompt `system`, since this will cause an issue with:

```
[{
  "role": "user",
  "message": "hello!"
}, {
  "role": "system",
  "message": "you are a happy panda"
}]
```